statistics/feedback.go

```diff
@@ -845,17 +845,12 @@ func logForIndex(prefix string, t *Table, idx *Index, ranges []*ranger.Range, ac
 		colName := idx.Info.Columns[rangePosition].Name.L
 		var rangeString string
 		// prefer index stats over column stats
-		if idx, ok := t.colName2Idx[colName]; ok {
-			if t.Indices[idx] == nil {
-				return
-			}
-			rangeString = logForIndexRange(t.Indices[idx], &rang, -1, factor)
+		if idxHist := t.indexStartWithColumnForDebugLog(colName); idxHist != nil {
+			rangeString = logForIndexRange(idxHist, &rang, -1, factor)
+		} else if colHist := t.columnByNameForDebugLog(colName); colHist != nil {
+			rangeString = colRangeToStr(colHist, &rang, -1, factor)
 		} else {
-			id := t.colName2ID[colName]
-			if t.Columns[id] == nil {
-				return
-			}
-			rangeString = colRangeToStr(t.Columns[t.colName2ID[colName]], &rang, -1, factor)
+			return
 		}
 		log.Debugf("%s index: %s, actual: %d, equality: %s, expected equality: %d, %s", prefix, idx.Info.Name.O,
```

Comment:
this line can be removed?

Reply:
Seems not?

Reply:
the `else` branch in line 852 returns the execution of this function. So this is a dead code.

Reply:
Oh, its meaning changed after i changing the code. This log should be in the if branch now.